File: src/main/java/net/serebryansky/routing/controller/RouteController.java
package net.serebryansky.routing.controller;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import lombok.val;
import net.serebryansky.routing.model.Route;
import net.serebryansky.routing.model.RouteSearchRequest;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

import static net.serebryansky.common.util.LoggingUtil.logOnEach;
import static org.springframework.http.ResponseEntity.ok;

@RestController
@RequestMapping("routes")
@Slf4j
@RequiredArgsConstructor
public class RouteController {
    @PostMapping("search")
    public ResponseEntity<Mono<Route>> search(@RequestBody RouteSearchRequest request) {
        val route = new Route();
        route.setPlaceStart(request.getPlaceStart());
        route.setPlaceEnd(request.getPlaceEnd());
        return ok(Mono.empty()
                .doOnEach(logOnEach(() -> log.info("Search route between {} and {}", request.getPlaceStart(), request.getPlaceEnd())))
                .then(Mono.just(route)));
    }
}

File: src/main/kotlin/net/serebryansky/routing/controller/RouteController.kt
package net.serebryansky.routing.controller

import mu.KotlinLogging
import net.serebryansky.routing.model.Route
import net.serebryansky.routing.model.RouteSearchRequest
import org.springframework.http.ResponseEntity
import org.springframework.http.ResponseEntity.ok
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("routes")
class RouteController {

    private val log = KotlinLogging.logger {}

    @PostMapping("search")
    suspend fun search(@RequestBody request: RouteSearchRequest): ResponseEntity<Route> {
        log.info { "Search route between ${request.placeStart} and ${request.placeEnd}" }
        val route = Route(
            placeStart = request.placeStart,
            placeEnd = request.placeEnd
        )
        return ok(route)
    }
}
